### PR TITLE
[10.0] [IMP] Contract adapted to purchase

### PR DESCRIPTION
**Pull request by ntsirintanis**
Backport of contract with recurring invoices from Odoo 11.0  to Odoo 10.0.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please stop closing and opening new PRs. That's not needed. You can add more commit to the existing PR, or even overwrite entire branch with `git push -f`.

**Comment by ntsirintanis** [comment]
@pedrobaeza My sincere apologies for all the noise, my very first pr turns out to be quite an adventure. @NL66278

**Comment by ntsirintanis** [comment]
@NL66278

File: contract/models/account_analytic_contract.py [hunk 1]
```diff
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
```
Comment (NL66278) [inline contract/models/account_analytic_contract.py]: Keep the coding line. Odoo 11.0 is on python > 3.0 so no coding line. Odoo 10.0 is still on python 2.7.x so coding line is needed.

File: contract/models/account_analytic_contract_line.py [hunk 2]
```diff
@@ -1,9 +1,9 @@
-# -*- coding: utf-8 -*-
```
Comment (NL66278) [inline contract/models/account_analytic_contract_line.py]: Keep coding line. Everywhere.

**Comment by ntsirintanis** [comment]
@NL66278  , @lfreeke , @gfcapalbo  please take a look.

**Review by NL66278 — APPROVED** [review]
:+1: LGTM

**Pull request closed without merging**